### ISSUE TITLE
Darkening when specific domains only

### ISSUE THREAD

**Issue by ALTanir** [rugk/website-dark-mode-switcher#12]
I want _web page to darker only specific domain, not for all as default_. So it needs a list to dark color for the linked page- which is listed. And others all are non-darken. For example(the system will be-)
Huge number of domains can visit by users. Yes, users who wants it always switching on/off for any site with their white list. And but some request is nearly same-that is, which users never need to use this darker software. But only for (suppose-) facebook, instagram, twitter, gmail, example.com1 ,2, and 3 (this 7 domains). He/she needs the list up. ### So two mode/group for list: 1)white list-not to darken & 2) **Off mode-** to perform darkening(for listed up) when switched off. And here, can disable option also be set.
Then users can disable extension or delete list as needs. I am all right? Thank you

**Comment by rugk** [comment]
Thanks for the idea.

I think the approach with a white & blacklist sounds fine. Though the UX could be harder, but I guess when you can right-click on the icon and choose "never darken this page" or "always darken this page", that would be okay, would not it? (and it saves this per page)

Also needs an "use default darkening options" menu entry – could be a radio button even.

**Comment by ALTanir** [comment]
Yes, and when addon is active, many many new sites will be launched unlimited. Then every sites have to list-up(you quoted) and if its not to visit anymore-it again needs to list out(not limited). So case to make users language by phrase and idioms for as it is like an aeroplane-driving. _For example- I need a glass of water and you suggest to bathing with hundreds of water and glass._  Main case was to launch an url, but driving will dismiss to case.
I have found a new idea by an other functional extension. That is an option called "invert white list". This always disabled self on every site. Here, if users need to enable- just right-button or toolbar icon click. Also ### this make the list once a huge mass within few months. Thus an user suppose needs not to keep this extension(**because unlimited clicks); yet requesting if possible**. Thank you

**Comment by mnaef** [comment]
First: Thanks for the Add on!

I use it (mostly) for everyday browsing. My System is always in dark mode but there are increasingly more website which offer a not too well implemented dark appearance. So I'd be very glad if the plugin would offer the option to store the preferred mode on a domain/website basis.

Best regards, Michael

**Comment by mnaef** [comment]
Hi rugk

Thanks for your elaborate reply. Too bad, maybe the browsers itself will se the necessity of such a feature and implement it themselfs. Thanks anyway. Still al lot better with you plugin that without.

**Comment by vamega** [comment]
I looked at doing this with the developer tools, and whatever API is being used by the developer tools, only toggles the light/dark preference for the current tab.

Am I correct in my understanding that that API is unavailable to Addons?

**Comment by rugk** [comment]
> Am I correct in my understanding that that API is unavailable to Addons?

From what I know, yes. It's [this API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/browserSettings/overrideContentColorScheme).

Opened https://bugzilla.mozilla.org/show_bug.cgi?id=1849786 now to clarify the request and maybe something will be done about that, otherwise it is still blocked.

**Comment by gaveitatry2** [comment]
> > Am I correct in my understanding that that API is unavailable to Addons?
> 
> From what I know, yes. It's [this API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/browserSettings/overrideContentColorScheme).
> 
> Opened https://bugzilla.mozilla.org/show_bug.cgi?id=1849786 now to clarify the request and maybe something will be done about that, otherwise it is still blocked.

Could you please reopen this Issue and not close it until it is resolved?

I found an extension called NightEye (https://nighteye.app) that unfortunately automatically applies dark mode globally to every website, but NightEye also allows you to toggle light/dark mode on each individual website by domain. So it IS possible to "darken only specific domains", if you can just figure out how they did it.

I do not recommend NightEye at all because their extension requires a paid subscription after a three month trial. And another reason I don't like NightEye is because it does the opposite of what I want. I want ALL websites to use light mode by default and I only want one or two websites to use dark mode. But to accomplish that in NightEye, I would have to change billions of websites to use light mode while keeping that one website that I want in dark mode. Too much work.

If you can figure out how NightEye lets users toggle light/dark mode per website by domain, you could give me and ALTanir the option to have all websites use light mode by default and just add certain websites to a "dark mode" list.

**Comment by gaveitatry2** [comment]
@ALTanir

I think that I finally figured out how to accomplish what I (and you) want.

1.) Install the Dark Reader extension (https://addons.mozilla.org/en-US/firefox/addon/darkreader/). It's probably available on Chrome and other browsers as well.

2.) Go to the website that you want to be in dark mode. Click on the Dark Reader toolbar button. Click on the 'Site List' tab. Click on 'Inverted Listed Only.' Press Alt + Shift + A to add the website that you are currently on.

This should make all websites use Light Mode (if that is what your browser is set to use), and Dark Reader will only work on the websites listed in the Inverted Listed Only section.

**Comment by rugk** [comment]
Indeed some.more replies to be added: This issue is not closed, it is open and blocked by API adjustments as said before.

Both NightyEye and DarkReader seem to work differently thorigh in a very basic way: they do somehow change websites with custom CSS/design changes.

This add-on has a specific aim and that us to only force a light or dark mode based on the website theme itself. So to would never have broken websites, because they offer this design themselves.

One could emulate that manually by not using the browser API intended for that. *However* I did (aka had to do) that in the past and it turned or to be hard and not a good idea due to the complexity involved.
See https://github.com/rugk/website-dark-mode-switcher/issues/22#issuecomment-988387763 for details and for a way to still use the old version though.

As said before in https://github.com/rugk/website-dark-mode-switcher/issues/55#issuecomment-1570302200 one could adjust that old version to add some blocklist or allowlists for specific domains, but well … it's old code and I am not in the position top maintain that.

---

What you can do is explain upstream aka Firefox in the opened Bugzilla ticket that you need that feature or just upvote it (you can do so there in the top part, it's somewhat hidden and cumbersome).

**Comment by gaveitatry2** [comment]
Oh, okay. So to summarize, this is only impossible by injecting CSS. But you prefer using the web browser API because it is less complicated. And we are hoping that a feature request will go through where the light/dark mode can be change on per website basis instead of globally. ?

I've since just created my own custom CSS for the site that I wanted in dark mode, so now I'm not using any extension. But for those people who want dark mode on individual websites, I would use Dark Reader. But it's too bad that no one is going to see my post on how to use Dark Reader because it has been marked and hidden as "off topic".

Anyway, thanks. I hope the feature request goes through. If it ever does, I might give it another try.
